Mathlib/Topology/Sheaves/Stalks.lean

```diff
@@ -115,6 +127,13 @@ theorem stalkFunctor_map_germ {F G : X.Presheaf C} (U : Opens X) (x : U) (f : F
     germ F x ≫ (stalkFunctor C x.1).map f = f.app (op U) ≫ germ G x :=
   colimit.ι_map (whiskerLeft (OpenNhds.inclusion x.1).op f) (op ⟨U, x.2⟩)
 
+@[reassoc, elementwise]
+lemma TopCat.Presheaf.germ_congr {C : Type*} [Category C] [HasColimits C] {X : TopCat}
+    (F : TopCat.Presheaf C X) {U : Opens X} (x x' : U) (h : x = x') :
+    F.germ x ≫ eqToHom (by rw [h]) = F.germ x' := by
+  subst h
+  simp
```

Comment:
I'm not sure we should have this. The intent is to only use `TopCat.Presheaf.stalkCongr`.

Reply:
The LHS can appear in proof terms though, even if using `TopCat.Presheaf.stalkCongr` as much as possible. So I think it is still useful, but I don't think I actually use it elsewhere in this PR. So if you insist, I remove it.

Reply:
Are there any examples? In theory we shouldn't have any `X.stalk a = X.stalk b` appearing in proof terms as they are equalities of types and are evil.

Reply:
I had one when preparing this PR, but later found a different solution, I can't find the example now. Let me remove this and if it reappears, I give you the example.

Re: No `X.stalk a = X.stalk b`. That is not what appears here though, right? The lemma is about comparing the maps `F.germ x` and `F.germ x'`.

Reply:
The `(by rw [h])` part is `X.stalk a = X.stalk b` I think.